### ISSUE TITLE
Total Count Rate in Spectrum Tree

### ISSUE THREAD

**Issue by Matt-Power** [Bruker-Amics/bugs_features#341]
## Uer Case 
Would be really useful to have a total count rate cut off in the Spectrum Tree - something along the lines of the BSE grouping but using the total counts collected (per point) instead. This would be really useful for dealing with things that are compositionally similar but one has a lot of light elements, contains microporosity, and a bunch of other things.

The ratio clustering can sort of be used but it is not very effective. 

## Solution 
1. Add a method to spectrum tree for total counts 
2. Calculate total counts of the segment from all x-rays in the segment 

## Acceptance Criteria 
- [ ] The user can use a spectrum tree step to separate based on total counts

**Comment by jonathan-knapp** [comment]
@MarkGuo7 do you understand this one? It would be super helpful in tough-up steps too to convert to the background for M4 measurements. I am wondering if the "psudoBSE" mode would address this right away and we could make that a feature in SEM too... What do you think?

**Comment by jonathan-knapp** [comment]
@matt, what about useing a filter in the spectrum tree what would result in two bins, (1) PASS and (2) FAIL

**Comment by Matt-Power** [comment]
It would be more versatile if you could have multiple bins as the result (rather than a simple pass / fail) - much the same way the BSE thing works e.g. selecting that option gives a plot of number of spectra in each total count bin (compared with n vs BSE brightness) and then you click on the chart to choose cut-offs. The resulting spectra are then grouped as usual e.g. 0 to 1000 total counts = background, 1000 to 1500 = mineral A, greater than 1500 = mineral B or whatever. The total counts would need to exclude the zero peak data which can mess things up.

**Comment by jonathan-knapp** [comment]
Wait... I think you can do this with count clustering already. Exactly like this. What is the difference?

**Comment by Matt-Power** [comment]
You'd think so but I've tried various combinations using count clustering and it doesn't work (or at least the time taken to make it work is just not worth it - and I've spent hours trying on occasion) and the step is also specific to the current sample. 
The issue is basically two-fold: 
1. The count clustering data in spectrum tree are normalised to 10000 so they don't easily match up with the total counts reported for each spectrum in the sample images tab. For instance, in the sample images window, taking an apatite spectrum at random, the total count value reported is 2895 (127 + 2768) but if you view all apatite spectra in the spectrum tree and select the entire chart (i.e. all counts), the reported value is 10,000 so you can't simply set a value that relates to the measurement e.g. if, based on observations in the measurement, I want to split into two groups - one Low Count apatite <2850  and one High Count apatite >2850, I can't just specify a cut at 2850. 
2. It is notionally possible to figure out the normalised value required from the reported total count, but you need to know the maximum total count for all of the spectra of interest (so that you have an idea of what the data are normalised to) but finding that spectrum takes some creativity and time in spectrum tree (i.e. filter to the highest value, mark the spectrum, then go to marked spectrum). Or you can do it by trial and error (which takes forever and is never successful). The normalisation also renders it impossible to make a step that will work for other samples as the data will be normalised to a different value for each sample (dependent on the maxiumum total count in the spectrum tree bin - which changes for each sample).

The above assumes that the normalisation is simply done to the highest total count of the bin which might well be an incorrect assumption. (Given the number of times I have tried to do this type of filtering without success (even with simple trial and error), I suspect that the process is indeed different to what I'm expecting and may well make it functionally impossible to do.)

Either way, being able to split based on the easily view-able total count value would be really useful for splitting several difficult mineral pairs, for removing background etc etc.

**Comment by AlexeiDolgolyov** [comment]
I will take it into progress after finishing current tasks

**Comment by AlexeiDolgolyov** [comment]
@knapp-amics , current count clusterring is really limited to 10000 border. Is it ok if I rename the existing method to `Normalized Count Clustering` (or not rename, because it's probably used as keyword) and add a new criteria `Total Count Clustering`.
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWowQ2c9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--521cedf9b2e02086241b267b6729de2e617509a4/image.png" alt="image.png" />
<img width="594" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeXIwQ2c9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--36143f4174da2cbf8d023ee0df4fe07e63e8c9be/image.png" alt="image.png" />

**Comment by AlexeiDolgolyov** [comment]
Seems like spectra is always normalized before stashing a spectrum tree. Will try to find an easy way to resolve this thing, but, probably, it might be really hard.

**Comment by jonathan-knapp** [comment]
Maybe there's another way around it. I think I know what Matt is trying to get out here and let me give it a thought today. Let's plan to push this one to 5.3 - I think that maybe we can find a way to do it with putting a total count rate check on the material definition itself

**Comment by jonathan-knapp** [comment]
@Matt-Power would a "total counts limit" work on the Reference Spectrum. This would prevent a match (is a requirement) if the total counts are above/below the specified limit. I would definitely want to have both a minimum and maximum value, and both optional.

**Comment by Matt-Power** [comment]
@knapp-amics yes, that should work. Main drawback is that it will require a reclassification each time you need to test (as opposed to almost instantaneous view if you could do in spectrum tree) but that's not really a big deal compared to the additional functionality it'll bring.

**Comment by jonathan-knapp** [comment]
Understood. 
@AlexeiDolgolyov it is likely the quickest for 5.2 to implement this as a limit. This is like the BSE or energy limits. We can enter a new task for fixing the spectrum tree normalization issue in 5.3. Unless... that is you have a better idea? Applying the limits is something we will want to do more of... Remember, we want to have more elemental criteria in 5.3...
